test/bfp_tests/src/bfp/real/test_bfp_s16_accumulate.c

```diff
@@ -45,12 +45,11 @@ TEST(bfp_s16_accumulate, bfp_s16_accumulate)
 
   double expected_f[MAX_LEN];
   int32_t DWORD_ALIGNED expected[MAX_LEN];
-  int32_t DWORD_ALIGNED actual[MAX_LEN];
+  int32_t DWORD_ALIGNED actual[MAX_LEN+1];
```

Comment:
why?

Reply:
> why?

To fix [ this problem](https://xmosjira.atlassian.net/jira/software/projects/AP/boards/79?assignee=557058%3A62a5d76c-12c2-47eb-b522-18f7e2bbbaea&selectedIssue=AP-169). We give an array of length 1023, but we are accessing value at index 1024. This test fails on XS3a too if we use the same input data as on Windows. An alternative solution would be to increase MAX_LEN from 1023 to 1024, but I am not sure if this is ok.

Reply:
@uvvpavel, I undid that change and I have updated the test. I made two changes:
  - I increased MAX_LEN from 127/1023 to 128/1024, to avoid the original invalid memory access
  - I increased the size of [acc](https://github.com/xmos/lib_xcore_math/blob/develop/test/bfp_tests/src/bfp/real/test_bfp_s16_accumulate.c#L42),  the array to store the accumulators by 1, to support the case in which the array size is equal to `MAX_LEN`.